### ISSUE TITLE
nodestream not supported with 3.3.0

### ISSUE THREAD

**Issue by emps** [Stuk/jszip#663]
as i use jszip@3.2.0 nodestream is working and is alsu supported with my angular 8
but as i update it to 3.3.0 nodestream become to false

**Comment by Stuk** [comment]
@mgechev as the author of the PR that went into 3.3.0 could you chime in here?

**Comment by mgechev** [comment]
Hey @Stuk, I'm lacking context; not sure what does this mean. Steps for reproduction would definitely help.

**Comment by emps** [comment]
just import in Angular Application and u will see that 3.2.0 has support for nodestream and 3.3.0 does not, if u cant do it by some reason let me know

**Comment by callmeguaishu** [comment]
got the same error.  JSzip.generateNodeStream() works well with JSZip 3.2.0, but failed with JSZip 3.3.0 & 3.4.0, throw an error: 'Error: nodestream is not supported by this platform', same as [491](https://github.com/Stuk/jszip/issues/491)

**Comment by callmeguaishu** [comment]
seems like readable-stream redirect to stream, may be some useful informiation here[776](https://github.com/DevExpress/devextreme-angular/issues/776)

**Comment by emps** [comment]
hope who broke it will fix it, i will keep 3.2.0 till then

**Comment by emaFusco** [comment]
We are having the same issue ('Error: nodestream is not supported by this platform') and we are stuck to version 3.2.0 as well.
Is there any news about this?

**Comment by stuarta0** [comment]
Yeah the PR https://github.com/Stuk/jszip/pull/614 breaks our build too; webpack+electron. I'm using patch-package to revert the change to [package.json](https://github.com/mgechev/jszip/commit/617e3ce439c1a0c694c2cb0a04f4a7578d94291f#diff-b9cfc7f2cdf78a7f4b91a753d10865a2) to fix it for now.

**Comment by jthrilly** [comment]
Just to add that we are also experiencing this issue with Electron.

**Comment by mgechev** [comment]
You can revert https://github.com/Stuk/jszip/pull/614 to get the previous behavior. Looks like doing so will break #524, #521, #477.

**Comment by jthrilly** [comment]
Sure, the workaround of patching the package.json does work, but I'm hoping we can come up with something more sustainable for Electron + webpack users. Seems like I found a fix, which is to use a resolve alias to force the node version of jszip to be bundled when targeting `electron-renderer`.

In your webpack config, add something like:

```
module.exports = {
  /// ... other config 
  alias: {
    jszip: require.resolve('jszip/lib/index.js'),
  }  
}
```

The fundamental problem is that webpack with the `electron-renderer` target will use the browser field in package.json to bundle the browser version of the library. This in turn will cause any JSZip method that calls `nodeStream()` to throw this error. 

On the surface this is perfectly logical - the nodeStream method isn't available in the browser, and the electron renderer is a modified chromium environment. However, node streams _are_ available in an Electron renderer environment when `nodeIntegration` is enabled. 

Not sure what can be done about the situation, the the above webpack config does resolve the issue.

**Comment by fuscoem** [comment]
Any update on this issue? Is it still downgrading to 3.2.2 the only solution to use generateNodeStream in a browser environment?

**Comment by vooit** [comment]
I have same experience in node 14...

**Comment by CodeCakePie** [comment]
Any update on this?
